File: software/examples/csharp/ExampleSimple.cs
using System;
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "XYZ"; // Change XYZ to the UID of your Real-Time Clock Bricklet

	static void Main()
	{
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletRealTimeClock rtc = new BrickletRealTimeClock(UID, ipcon); // Create device object

		ipcon.Connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current date and time
		int year; byte month, day, hour, minute, second, centisecond, weekday;
		rtc.GetDateTime(out year, out month, out day, out hour, out minute, out second,
		                out centisecond, out weekday);

		Console.WriteLine("Year: " + year);
		Console.WriteLine("Month: " + month);
		Console.WriteLine("Day: " + day);
		Console.WriteLine("Hour: " + hour);
		Console.WriteLine("Minute: " + minute);
		Console.WriteLine("Second: " + second);
		Console.WriteLine("Centisecond: " + centisecond);

		if(weekday == BrickletRealTimeClock.WEEKDAY_MONDAY)
		{
			Console.WriteLine("Weekday: Monday");
		}
		else if(weekday == BrickletRealTimeClock.WEEKDAY_TUESDAY)
		{
			Console.WriteLine("Weekday: Tuesday");
		}
		else if(weekday == BrickletRealTimeClock.WEEKDAY_WEDNESDAY)
		{
			Console.WriteLine("Weekday: Wednesday");
		}
		else if(weekday == BrickletRealTimeClock.WEEKDAY_THURSDAY)
		{
			Console.WriteLine("Weekday: Thursday");
		}
		else if(weekday == BrickletRealTimeClock.WEEKDAY_FRIDAY)
		{
			Console.WriteLine("Weekday: Friday");
		}
		else if(weekday == BrickletRealTimeClock.WEEKDAY_SATURDAY)
		{
			Console.WriteLine("Weekday: Saturday");
		}
		else if(weekday == BrickletRealTimeClock.WEEKDAY_SUNDAY)
		{
			Console.WriteLine("Weekday: Sunday");
		}

		// Get current timestamp
		long timestamp = rtc.GetTimestamp();
		Console.WriteLine("Timestamp: " + timestamp + " ms");

		Console.WriteLine("Press enter to exit");
		Console.ReadLine();
		ipcon.Disconnect();
	}
}

File: software/examples/java/ExampleSimple.java
import com.tinkerforge.IPConnection;
import com.tinkerforge.BrickletRealTimeClock;
import com.tinkerforge.BrickletRealTimeClock.DateTime;

public class ExampleSimple {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;

	// Change XYZ to the UID of your Real-Time Clock Bricklet
	private static final String UID = "XYZ";

	// Note: To make the example code cleaner we do not handle exceptions. Exceptions
	//       you might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletRealTimeClock rtc = new BrickletRealTimeClock(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Get current date and time
		DateTime dateTime = rtc.getDateTime(); // Can throw com.tinkerforge.TimeoutException

		System.out.println("Year: " + dateTime.year);
		System.out.println("Month: " + dateTime.month);
		System.out.println("Day: " + dateTime.day);
		System.out.println("Hour: " + dateTime.hour);
		System.out.println("Minute: " + dateTime.minute);
		System.out.println("Second: " + dateTime.second);
		System.out.println("Centisecond: " + dateTime.centisecond);

		if(dateTime.weekday == BrickletRealTimeClock.WEEKDAY_MONDAY) {
			System.out.println("Weekday: Monday");
		} else if(dateTime.weekday == BrickletRealTimeClock.WEEKDAY_TUESDAY) {
			System.out.println("Weekday: Tuesday");
		} else if(dateTime.weekday == BrickletRealTimeClock.WEEKDAY_WEDNESDAY) {
			System.out.println("Weekday: Wednesday");
		} else if(dateTime.weekday == BrickletRealTimeClock.WEEKDAY_THURSDAY) {
			System.out.println("Weekday: Thursday");
		} else if(dateTime.weekday == BrickletRealTimeClock.WEEKDAY_FRIDAY) {
			System.out.println("Weekday: Friday");
		} else if(dateTime.weekday == BrickletRealTimeClock.WEEKDAY_SATURDAY) {
			System.out.println("Weekday: Saturday");
		} else if(dateTime.weekday == BrickletRealTimeClock.WEEKDAY_SUNDAY) {
			System.out.println("Weekday: Sunday");
		}

		// Get current timestamp
		long timestamp = rtc.getTimestamp(); // Can throw com.tinkerforge.TimeoutException
		System.out.println("Timestamp: " + timestamp + " ms");

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}
